andela/travel_tool_front
# #161147524 Remove edit restriction of guest house centres

#### What does this PR do?
 This PR addresses the removal of the editing restriction on a guest house center, fix edit link and removing new room row when the component is mounted on the edit form modal.
 
#### Description of Task to be completed?
- Remove the restriction of editing guest house by the travel admin who created it
- Add edit icon to the edit link
- Prevent new room row from being added when `componentDidMount` is called on editing guest house data.

#### How should this be manually tested?
- Clone the repository
- Run `make start` docker build command on the terminal to install all dependencies and run the App in the docker development environment server.
- On the browser, type the localhost URL ` http://localhost:3000` and Login with your Andela email account into the App.
- On the Requests page, click on the `Residence` link on the side navigation bar and click on `Manage` and the accommodation page should be displayed.
- Click on a guest house center from the available list of `seeded` guest house centers available on the page.
- Clicking on a guest house center should open a detail page showing details of the guest house center.
- On the guest house details page, click on the `Edit Guest House` link which has the edit icon attached to it and it opens a modal form.
- Edit the guest house center data accordingly and click on `save changes` button.
- The guest house center data is updated as expected and displayed on the guest house details page

 #### Any background context you want to provide?
- Make sure that you are logged in as a `Super admin` or `Travel admin` in order for you to access the `Accommodation page`
- In order to successfully edit guest house center that was seeded, please change the `beds`, `rooms` and `guest house` database `IDs` to numbers from 10 and above. This is to ensure that there are no conflicts.

#### What are the relevant pivotal tracker stories?
- [161147524](https://www.pivotaltracker.com/story/show/161147524)

#### Screenshots (if appropriate)
Screenshot showing the edit icon.

<img width="1273" alt="screen shot 2018-10-12 at 16 35 53" src="https://user-images.githubusercontent.com/12033494/46872816-dd8afb00-ce3d-11e8-8421-e8f31fccdc8a.png">

Screenshot showing modal form without new room row added when modal opens

<img width="1280" alt="screen shot 2018-10-12 at 16 55 17" src="https://user-images.githubusercontent.com/12033494/46873616-b6352d80-ce3f-11e8-9ece-4251f980f968.png">




#### Questions:
- None
